x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/GetDataFrameAnalyticsStatsAction.java

```diff
@@ -202,6 +206,9 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             public XContentBuilder toUnwrappedXContent(XContentBuilder builder) throws IOException {
                 builder.field(DataFrameAnalyticsConfig.ID.getPreferredName(), id);
                 builder.field("state", state.toString());
+                if (failureReason != null) {
+                    builder.field("failure_reason", failureReason);
```

Comment:
I think it may be prudent to use a more generic `reason` as the field name. If the `state: failed`, then we know it failed. If we ever decide to populate `reason` with other information to indicate `upgrading`, `stopping`, etc. it would be good to be able to use the same field.

Reply:
But in the short term while we only populate the field when the job is `failed` we might get questions about why `reason` is empty most of the time.

For the question of why it's not assigned to a node during an upgrade this will already be available in `assignment_explanation`.

I guess the question is whether to combine `assignment_explanation` and `failure_reason` into a single `reason` field.  I'm happy to keep them separate.

Reply:
My thinking was to avoid having a `state` object. And as the stats object already has `assignment_explanation`, `reason` seems to be confusing. If we need to add more things in the future I think we'll have to rethink the object structure a bit.